src/ospjolt/activescene/joltinteg_fn.cpp

```diff
@@ -0,0 +1,254 @@
+            //As we are translating the whole world, we don't need to wake up asleep bodies. 
+            bodyInterface.SetPosition(bodyId, position, EActivation::DontActivate);
+        }
+    }
+}
+
+using Corrade::Containers::ArrayView;
+
+void SysJolt::update_world(
+        ACtxPhysics&                rCtxPhys,
+        ACtxJoltWorld&              rCtxWorld,
+        float                       timestep,
+        ACompTransformStorage_t&    rTf) noexcept
+{
+    PhysicsSystem *pJoltWorld = rCtxWorld.m_world.get();
+    BodyInterface &bodyInterface = pJoltWorld->GetBodyInterface();
+
+    // Apply changed velocities
+    for (auto const& [ent, vel] : std::exchange(rCtxPhys.m_setVelocity, {}))
+    {
+        OspBodyId const ospBodyId     = rCtxWorld.m_entToBody.at(ent);
+        JoltBodyId const bodyId     = rCtxWorld.m_ospToJoltBodyId[ospBodyId];
+
+        bodyInterface.SetLinearVelocity(bodyId, Vec3(vel.x(), vel.y(), vel.z()));
+    }
+
+    rCtxWorld.m_pTransform = std::addressof(rTf);
+
+    // Update the world
+    pJoltWorld->Update(timestep, 1, &rCtxWorld.m_temp_allocator, rCtxWorld.m_joltJobSystem.get());
+}
+
+void SysJolt::remove_components(ACtxJoltWorld& rCtxWorld, ActiveEnt ent) noexcept
+{
+    auto itBodyId = rCtxWorld.m_entToBody.find(ent);
+
+    if (itBodyId != rCtxWorld.m_entToBody.end())
+    {
+        OspBodyId const bodyId = itBodyId->second;
+        rCtxWorld.m_bodyIds.remove(bodyId);
+        rCtxWorld.m_bodyToEnt[bodyId] = lgrn::id_null<ActiveEnt>();
+        rCtxWorld.m_entToBody.erase(itBodyId);
+    }
+
+}
+
+TransformedShapePtr_t SysJolt::create_primitive(ACtxJoltWorld &rCtxWorld, osp::EShape shape)
+{
+    Shape* joltShape;
+    switch (shape)
+    {
+    case EShape::Sphere:
+        joltShape = (Shape *) new SphereShape(1.0f);
+        break;
+    case EShape::Box:
+        joltShape = (Shape *) new BoxShape(Vec3Arg(1.0f, 1.0f, 1.0f));
+        break;
+    case EShape::Cylinder:
+        //cylinder needs to be internally rotated 90° to match with graphics
+        joltShape = (Shape *) new CylinderShape(1.0f, 2.0f);
+        joltShape = (Shape *) new RotatedTranslatedShape(Vec3Arg::sZero(), Quat::sRotation(Vec3::sAxisX(), JPH_PI/2), joltShape);
+        break;
+    default:
+        // TODO: support other shapes, sphere is used for now
+        joltShape = (Shape *) new SphereShape(1.0f);
+        break;
+    }
+    return std::make_unique<TransformedShape>(RVec3::sZero(), Quat::sZero(), joltShape, BodyID(), SubShapeIDCreator());
+}
+
+void SysJolt::orient_shape(TransformedShapePtr_t& pJoltShape, osp::EShape ospShape, osp::Vector3 const &translation, osp::Matrix3 const &rotation, osp::Vector3 const &scale)
+{
+    auto rawQuat = Magnum::Quaternion::fromMatrix(rotation).data();
+    Quat joltRotation(rawQuat[0], rawQuat[1], rawQuat[2], rawQuat[3]);
+
+    //scale the shape directly
+    pJoltShape->mShape = pJoltShape->mShape->ScaleShape(Vec3Arg(scale.x(), scale.y(), scale.z())).Get();
```

Comment:
Is the compound shape (collider) transform supposed to be absolute ? I assumed it was relative to the compound shape transform (mostly because the recursive function starts with identity transform)
That is, the final absolute transform needs to be "compound shape transform" * "relative subshape transform".

I do agree that there are better ways to handle this though. I'll scrap orient_shape and create_primitive to make something that only adds ScaledShape wrapper when it is needed (ie when we are not creating the shape in the same spot, and when the shape is not already scaled).